### DISCUSSION TITLE
RustDesk Console Pro Version - List Hostnames along with IDs / Import to Address Book

### DISCUSSION THREAD

**Discussion by cm-quint** [rustdesk/rustdesk#5007] [Feature Request]
Hi everyone,

Just onboarded the Pro version yesterday to support this great app. One thing that I think it could be of great help to organize the computers as they start connecting to the server would be to also have the hostnames instead of only the ID in the Devices list.

Or, another idea would be to auto-populate the Address Book with the computers that a user has access to (either automatically or through an 'import' feature).

**Comment by dinger1986** [comment]
Both sound like great ideas!

**Comment by SergeyMy** [comment]
all your clients are registered on YOUR rendezvous server. also all your clients are marked with their ID (every 30 seconds) on your api_server

**Reply by cm-quint** [reply]
We know that, but it's of no use to help identifying the host if you don't previously know the hostname/or do a first access. We're a MSP migrating from Teamviewer and knowing the hostname first hand and having the host automatically added to an address book helps a lot organizing things. Some users are very limited, sometimes being hard to even instruct them to download and execute a file.

**Reply by dinger1986** [reply]
yes totally agree this should really be on rustdesk-server-pro though :)

**Comment by rustdesk** [comment]
> either automatically or through an 'import' feature

@21pages let's add an option of this on addressbook page. Let's add "import" or "sync" to the menu first. I think "sync" better, because sometimes, the user may change password, we need use "sync" to refresh addressbook.

**Reply by 21pages** [reply]
Ok, it's like the group, alrougth it's buggy now. Address book need sync too.

**Reply by dinger1986** [reply]
host names show in the addressbook, 

![image](https://github.com/rustdesk/rustdesk/assets/7244447/09f354c7-db2c-4b51-af8b-173f9ea77d8e)

Would be good for them to show in the server like this
![image](https://github.com/rustdesk/rustdesk/assets/7244447/77ba0a08-a22d-40d4-b517-e8da2d2a41ef)

**Reply by ghost** [reply]
Sync is not a good idea based on [this](https://github.com/rustdesk/rustdesk-server-pro/issues/34)

Import yes, sync no.

**Reply by rustdesk** [reply]
Our sync will only add, not removing.

**Comment by it-boerner** [comment]
I also think that the address book needs to be synchronized with this option

**Comment by SergeyMy** [comment]
1. each client on **hbbs** leaves its own UUID
2. as well as everyone who logged in!!! leaves its UUID via **web/api**
:)
here is the link for the contact (address book) UUID+ID+Login
just thoughts out loud :shipit:

**Comment by ghost** [comment]
+1 for hostnames. Far easier to read than random ID numbers.
